developer_guides/topology2/topology2.rst

```diff
@@ -0,0 +1,895 @@
+
+.. code-block:: bash
+
+	DefineAttribute."ramp_step_type" {
+		type	"string"
+		# Token set reference name
+		token_ref	"sof_tkn_volume.word"
+		constraints {
+			!valid_values [
+				"linear"
+				"log"
+				"linear_zc"
+				"log_zc"
+			]
+			!tuple_values [
+				0
+				1
+				2
+				3
+			]
+		}
+	}
+
+1.1.7 A complete class definition
+*********************************
+
+Puting it all together, the complete defintiion for the pga widget class is as follows:
+
+.. code-block:: bash
+
+	Class.Widget."pga" {
+		# attribute definitions
+		DefineAttribute.instance {
+			type	integer
+		}
+		DefineAttribute.index {
+			type	integer
+		}
+		DefineAttribute."type" {
+			type	"string"
+		}
+		DefineAttribute."uuid" {
+			type	"string"
+			# Token set reference name and type
+			token_ref	"sof_tkn_comp.uuid"
+		}
+		DefineAttribute."preload_count" {}
+
+		#
+		# Volume ramp step in milliseconds
+		#
+		DefineAttribute."ramp_step_ms" {
+			# Token set reference name
+			token_ref	"sof_tkn_volume.word"
+			constraints {
+				min 200
+				max 500
+			}
+		}
+		DefineAttribute."ramp_step_type" {
+			type	"string"
+			# Token set reference name
+			token_ref	"sof_tkn_volume.word"
+			constraints {
+				!valid_values [
+					"linear"
+					"log"
+					"linear_zc"
+					"log_zc"
+				]
+				!tuple_values [
+					0
+					1
+					2
+					3
+				]
+			}
+		}
+		
+		# attribute qualifiers
+		attributes {
+			#
+			# The PGA widget name would be constructed using the index and instance
+			# attributes. For ex: "pga.1.1" or "pga.10.2" etc.
+			#
+			!constructor [
+				"index"
+				"instance"
+			]
+
+			#
+			# immutable attributes cannot be modified in the object instance
+			#
+			!immutable [
+				"uuid"
+				"type"
+			]
+
+			#
+			# deprecated attributes should not be added in the object instance
+			#
+			!deprecated [
+				"preload_count"
+			]
+
+			#
+			# pga widget objects instantiated within the same alsaconf node must have
+			# unique instance attribute
+			#
+			unique	"instance"
+		}
+
+		# default attribute values
+		type 		"pga"
+		uuid 		"7e:67:7e:b7:f4:5f:88:41:af:14:fb:a8:bd:bf:86:82"
+		ramp_step_ms	200
+	}
+
+1.2 Objects
+***********
+Objects are used to instantiate multiple instances of the same class to avoid duplicating
+common attribute definitions. Objects are instantiated with the new keyword "Object" followed by
+3 nodes in order as follows:
+
+.. code-block:: bash
+
+	Object.Widget.pga."1" {}
+
+The nodes refer to the following:
+
+* Class group to which the object's class belongs i.e. "Widget"
+* Class name i.e. "pga"
+* Unique attribute value: This is the value for the attribute that is qualified as "unique" in the
+  class definition i.e. "instance"
+
+Using the pga class definition above, a pga widget object can be declared as follows:
+
+.. code-block:: bash
+
+	Object.Widget.pga."1" {
+		index 5
+	}
+
+where 1 is the value for the unique attribute, "instance", in the pga class definition and the
```

Comment:
1 is the value for the unique attribute?

the only thing unique above is 
unique "instance" 

but instance is 5, so what is the 1?